src/builtins/core/duration/normalized.rs

```diff
@@ -776,146 +776,133 @@ impl NormalizedDurationRecord {
+        for unit in UNIT_VALUE_TABLE[largest_unit_index..smallest_unit_index]
+            .iter()
+            .rev()
+            .copied()
+        {
+            // b. If unit is not week, or largestUnit is week, then
+            if unit != Unit::Week || largest_unit == Unit::Week {
+                let end_duration = match unit {
+                    // i. If unit is year, then
+                    Unit::Year => {
+                        // 1. Let years be duration.[[Date]].[[Years]] + sign.
+                        let years = self
                             .date()
                             .years
                             .checked_add(sign.as_sign_multiplier().into())
-                            .ok_or(TemporalError::range())?,
-                        0,
-                        0,
-                        0,
-                    )?
-                }
-                // ii. Else if unit is "month", then
-                Unit::Month => {
-                    // 1. Let months be duration.[[Months]] + sign.
-                    // 2. Let endDuration be ? CreateNormalizedDurationRecord(duration.[[Years]], months, 0, 0, ZeroTimeDuration()).
-                    DateDuration::new(
-                        duration.date().years,
-                        duration
+                            .ok_or(TemporalError::range())?;
+
+                        // 2. Let endDuration be ? CreateDateDurationRecord(years, 0, 0, 0).
+                        DateDuration::new(years, 0, 0, 0)?
+                    }
+                    // ii. Else if unit is month, then
+                    Unit::Month => {
+                        // 1. Let months be duration.[[Date]].[[Months]] + sign.
+                        let months = self
                             .date()
                             .months
                             .checked_add(sign.as_sign_multiplier().into())
-                            .ok_or(TemporalError::range())?,
-                        0,
-                        0,
-                    )?
-                }
-                // iii. Else if unit is "week", then
-                Unit::Week => {
-                    // 1. Let weeks be duration.[[Weeks]] + sign.
-                    // 2. Let endDuration be ? CreateNormalizedDurationRecord(duration.[[Years]], duration.[[Months]], weeks, 0, ZeroTimeDuration()).
-                    DateDuration::new(
-                        duration.date().years,
-                        duration.date().months,
-                        duration
+                            .ok_or(TemporalError::range())?;
+
+                        // 2. Let endDuration be ? AdjustDateDurationRecord(duration.[[Date]], 0, 0, months).
+                        duration.date().adjust(0, Some(0), Some(months))?
+                    }
+                    // iii. Else,
+                    unit => {
+                        // 1. Assert: unit is week.
+                        temporal_assert!(unit == Unit::Week);
```

Comment:
thought (non-blocking): should we use `temporal_assert` over `debug_assert` in the situations where the abstract is pretty far down the code path?

Really, not sure. I'm fine with merging this as is. But we may want to revisit `temporal_assert` vs `debug_assert` in the future for optimization purposes in production builds. For reference, I believe SM uses MOZ_ASSERT for these assertions, which i think mimics the `debug_assert` functionality

Reply:
Hmmm.. not really sure either, maybe we could take a general approach we assert only if omitting the assertion can cause a panic or enter some weird unrecoverable state, otherwise `debug_assert`.

Though the debug asserts have a small disadvantage that if you don't build the test code/tester with `relwithdebug` you won't see it, but this point is minor.

Reply:
Yeah, the case with this is that it's specifically an internal method. So if this is called with the asserted values, we are either not to specification or the spec has an issue. So I think it's fine to live with not having notice in the production code if we have unit tests that can be run in debug.